### ISSUE TITLE
`public` resources should not set redirect_to cookie

### ISSUE THREAD

**Issue by silverwind** [go-gitea/gitea#11088]
`/js/userheatmap.js` is currently requested by the serviceworker which is a bug in itself but the request causes a bad cookie to be set when authentication is enabled:

```
Request URL: /js/userheatmap.js
Request Method: GET
```
```
status: 302
location: /user/login
set-cookie: redirect_to=%2Fjs%2Fuserheatmap.js; Path=/
````

The next time the user logs in with that cookie set, they will be redirected to /js/userheatmap.js which causes a page loading error. We should just 404 any requests that would be routed to a file in `public`, or at least the js/css/img/vendor folders.

**Comment by guillep2k** [comment]
Fixed in #11091

**Comment by silverwind** [comment]
No, not fixed, it's just a workaround. Proper fix will be in the router, I plan to look into it.
